src/analysis/modals/ComputeModal/AzureComputeModal/AzureComputeModal.js

```diff
@@ -83,8 +91,8 @@ export const AzureComputeModalBase = ({
         autopauseThreshold: runtimeDetails ? runtimeDetails.autopauseThreshold || autopauseDisabledValue : defaultAutopauseThreshold,
       });
     });
-    loadCloudEnvironment();
-  });
+    refreshRuntime();
+  }, [currentRuntime, location, onError, workspaceId, setCurrentRuntimeDetails, setComputeConfig]);
```

Comment:
+1000 for `useEffect` with dependencies vs `useOnMount`.

Though setters returned from `useState` are stable and don't need to be included in dependencies.

```suggestion
  }, [currentRuntime, location, onError, workspaceId]);
```